diff --git a/src/meson.build b/src/meson.build
index 8a86f9e..7f4e5f7 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -473,6 +473,7 @@ if introspection
          install_dir_gir: girdir,
      install_dir_typelib: typelibdir,
               extra_args: libadwaita_gir_extra_args,
+            dependencies: [dependency('girepository-2.0'), dependency('gtk4')],
    )
 
    if get_option('vapi')
